docs/IntegrationWithExistingApps.md

```diff
@@ -720,6 +720,8 @@ public boolean onKeyUp(int keyCode, KeyEvent event) {
 
 That's it, your activity is ready to run some JavaScript code.
 
+> If your app is targeting the Android `api level 23` or greater, make sure you have, for the development build, the `overlay permission` enabled. You can check it with `Settings.canDrawOverlays(this);`. This is required because, if your app produces an error in the react native component, the error view is displayed above all the other windows. Due to the new permissions system, introduced in the api level 23, the user needs to approve it.
```

Comment:
> The template shipped with RN has the target SDK set to 22, so most developer won't face any issue unless they manually set the target SDK to 23. Probably the only thing we need to document is that you should target API level 22 until we officially support it.

If you are integrating with an existent app (thats the point of this guide) you should be targeting api level 24. See the [bot](https://twitter.com/targtSdkVersion). So, everyone that follow that guide probably will be targeting `>22`.

>Also I don't think it's something to do with if people use an emulator or not. It should crash anyways.

For the `Hello World`  described in this guide, I think it will only crash in the device, since the emulator is in the localhost and will be able to work without any issue.

> In case of overlay permissions, RN should show a dialog saying the permission is not granted, instead of crashing. 

That would be great! 

> Currently it does redirect you to the permissions screen and then crashes, so probably a minor thing we need to fix.

At least on my device I only see the crash. It does not redirect me to the permissions screen.

Like I said before, IMO, the disclaimer will help other developers to perform a seamless integration with the existing apps.

@satya164 thanks again for the feedback and for the clarification.